emacs/config.org
#+TITLE: Emacs configuration file 
#+AUTHOR: Jasper Reuhman
#+STARTUP: showeverything
#+OPTIONS: toc:2

* Table of contents :toc:
- [[#plug-in-manager][PLUG-IN MANAGER]]
  - [[#elpaca][Elpaca]]
- [[#plugins][PLUGINS]]
  - [[#dashboard][Dashboard]]
  - [[#evil-mode][Evil-mode]]
  - [[#general][General]]
  - [[#ivy][Ivy]]
  - [[#magit][Magit]]
  - [[#which-key][Which Key]]
- [[#org-mode][ORG Mode]]
  - [[#table-of-contents][Table of Contents]]
  - [[#nicer-bullets][Nicer bullets]]
  - [[#source-code-block-expansion][Source Code Block expansion]]
  - [[#org-roam][ORG Roam]]
- [[#theme][Theme]]
  - [[#transparency][Transparency]]
  - [[#add-all-the-icons][Add all the icons]]
  - [[#color-theme][Color theme]]
- [[#general-emacs-settings][General Emacs settings]]

* PLUG-IN MANAGER 
** Elpaca
#+begin_src emacs-lisp
    (defvar elpaca-installer-version 0.11)
    (defvar elpaca-directory (expand-file-name "elpaca/" user-emacs-directory))
    (defvar elpaca-builds-directory (expand-file-name "builds/" elpaca-directory))
    (defvar elpaca-repos-directory (expand-file-name "repos/" elpaca-directory))
    (defvar elpaca-order '(elpaca :repo "https://github.com/progfolio/elpaca.git"
    				    :ref nil :depth 1 :inherit ignore
    				    :files (:defaults "elpaca-test.el" (:exclude "extensions"))
    				    :build (:not elpaca--activate-package)))
    (let* ((repo  (expand-file-name "elpaca/" elpaca-repos-directory))
    	    (build (expand-file-name "elpaca/" elpaca-builds-directory))
    	    (order (cdr elpaca-order))
	    (default-directory repo))
	(add-to-list 'load-path (if (file-exists-p build) build repo))
    	(unless (file-exists-p repo)
    	(make-directory repo t)
    	(when (<= emacs-major-version 28) (require 'subr-x))
    	(condition-case-unless-debug err
    	    (if-let* ((buffer (pop-to-buffer-same-window "*elpaca-bootstrap*"))
    			((zerop (apply #'call-process `("git" nil ,buffer t "clone"
    							,@(when-let* ((depth (plist-get order :depth)))
    							    (list (format "--depth=%d" depth) "--no-single-branch"))
    							,(plist-get order :repo) ,repo))))
    			((zerop (call-process "git" nil buffer t "checkout"
    					    (or (plist-get order :ref) "--"))))
    			(emacs (concat invocation-directory invocation-name))
    			((zerop (call-process emacs nil buffer nil "-Q" "-L" "." "--batch"
    					    "--eval" "(byte-recompile-directory \".\" 0 'force)")))
    			((require 'elpaca))
    			((elpaca-generate-autoloads "elpaca" repo)))

    		(progn (message "%s" (buffer-string)) (kill-buffer buffer))
    		(error "%s" (with-current-buffer buffer (buffer-string))))
    	    ((error) (warn "%s" err) (delete-directory repo 'recursive))))
    	(unless (require 'elpaca-autoloads nil t)
    	(require 'elpaca)
    	(elpaca-generate-autoloads "elpaca" repo)
    	(load "./elpaca-autoloads")))
    (add-hook 'after-init-hook #'elpaca-process-queues)
    (elpaca `(,@elpaca-order))

    (elpaca elpaca-use-package
    (elpaca-use-package-mode))

#+end_src

* PLUGINS
** Dashboard
#+begin_src emacs-lisp 
  (use-package dashboard
    :ensure t
    :config
    (dashboard-setup-startup-hook))
#+end_src

** Evil-mode 

Because VIM-keys are just better!

#+begin_src emacs-lisp
  (use-package evil
  :ensure t
  :demand t
  :init
  (setq evil-want-C-i-jump nil)
  (evil-mode))
#+end_src

** General
#+begin_src emacs-lisp
  (use-package general
  :ensure t
  :demand t
  :config
  (general-evil-setup)

  (general-create-definer jr/leader-keys
    :states '(normal insert visual emacs)
    :keymap 'override
    :prefix "SPC"
    :global-prefix "M-SPC")

  (jr/leader-keys
   "." '(find-file :wk "Find file"))
  )
#+end_src

** Ivy
#+begin_src emacs-lisp
  (use-package ivy
    :ensure t
    :bind
    (("C-c C-r" . ivy-resume)
     ("C-x B" . ivy-switch-buffer-other-window))
    :custom
    (setq ivy-use-virtual-buffers t)
    (setq ivy-count-format "(%d/%d) ")
    (setq enable-recursive-minibuffers t)
    :config
    (ivy-mode))

  (use-package all-the-icons-ivy-rich
    :ensure t
    :init (all-the-icons-ivy-rich-mode 1))

  (use-package counsel
    :ensure t
    :after ivy
    :config (counsel-mode))

  (use-package ivy-rich
    :ensure t
    :after ivy
    :init (ivy-rich-mode 1)
    :custom
    (ivy-virtual-abbreviate 'full
     ivy-rich-switch-buffer-align-virtual-buffer t
     ivy-rich-path-style 'abbrev))
 #+end_src

** Magit

We also install transient here, since this is required by magit. The version shipped with
my emacs is older than required.

#+begin_src emacs-lisp
  (use-package transient
    :ensure t)

  (use-package magit
    :after transient
    :ensure t)
#+end_src

** Which Key
#+begin_src emacs-lisp 
  (use-package which-key
    :ensure t
    :config
    (which-key-mode 1))
#+end_src

* ORG Mode
org-mode specific plugins

** Table of Contents
#+begin_src emacs-lisp
  (use-package toc-org
  :ensure t
  :commands toc-org-enable
  :init
  (add-hook 'org-mode-hook 'toc-org-mode)
  )
#+end_src

** Nicer bullets
#+begin_src emacs-lisp
      (use-package org-bullets
        :ensure t
        :init
      (add-hook 'org-mode-hook (lambda() (org-bullets-mode 1))))
#+end_src

** Source Code Block expansion
#+begin_src emacs-lisp
    (require 'org-tempo)
#+end_src

** ORG Roam
#+begin_src emacs-lisp
  (use-package org-roam
    :ensure t
    :custom
    (org-roam-directory "~/RoamNotes")
    :bind( ("C-c n l" . org-roam-buffer-toggle)
  	 ("C-c n f" . org-roam-node-find)
  	 ("C-c n i" . org-roam-node-insert))
    :config
    (org-roam-setup))
#+end_src

* Theme
** Transparency
#+begin_src emacs-lisp
(add-to-list 'default-frame-alist '(alpha-background . 90))
#+end_src

** Add all the icons
#+begin_src emacs-lisp
  (use-package all-the-icons
  :ensure t
  :if (display-graphic-p))

  (use-package all-the-icons-dired
  :ensure t
  :hook (dired-mode . (lambda () (all-the-icons-dired-mode t))))
#+end_src

** Color theme
#+begin_src emacs-lisp
  (use-package doom-themes
  :ensure t
  :config
  (setq doom-themes-enable-bold t
        doom-themes-enable-italic t)
  (load-theme 'doom-one t)

  (doom-themes-visual-bell-config)
  (doom-themes-org-config))
#+end_src

* General Emacs settings
#+begin_src emacs-lisp
  (setq backup-directory-alis '((".*" . "~/.local/share/Trash/files")))

  (setq inhibit-startup-screen t)

  (menu-bar-mode -1)
  (tool-bar-mode -1)
  (scroll-bar-mode -1)

  (global-display-line-numbers-mode 1)
  (setq display-line-numbers-type 'relative)

  (global-visual-line-mode t)

  (set-face-attribute 'default nil
      :font "SpaceMono Nerd Font"
      :height 150
      :weight 'normal) 
#+end_src


qtile/config.org
#+AUTHOR: Jasper Reuhman
#+TITLE: Qtile configuration file
#+PROPERTY: header-args :tangle config.py

* Table of contents :toc:
- [[#main-imports][Main imports]]
- [[#variables][Variables]]
- [[#keymappings][Keymappings]]
- [[#groups][Groups]]
- [[#scratchpads][Scratchpads]]
- [[#layouts][Layouts]]
- [[#top-bar][Top Bar]]

* Main imports
#+BEGIN_SRC python
import os
import subprocess

from libqtile import bar, layout, qtile, widget, hook
from libqtile.config import Click, Drag, Group, Key, Match, Screen, KeyChord, DropDown, ScratchPad
from libqtile.lazy import lazy
from libqtile.utils import guess_terminal
from colors import *
#+END_SRC

* Variables

The mod-key we use is the Windows Key.

#+BEGIN_SRC python
mod = "mod4"

theme=RofiOne
terminal = "kitty"
browser = "brave" 
rofi = "rofi -show combi"

def run_script(qtile, script):
    run_command = os.path.expanduser('~/.config/qtile/scripts/' + script)
    subprocess.call(run_command)

@hook.subscribe.startup_once
def autostart_once():
    #lazy.spawn("picom -bc")
    #lazy.spawn("emacs --daemon")
    autostart_script = os.path.expanduser('~/.config/qtile/scripts/autostart.sh')
    subprocess.call(autostart_script)

#+END_SRC

* Keymappings

#+BEGIN_SRC python
keys = [
    Key([mod], "h", lazy.layout.left(), desc="Move focus to left"),
    Key([mod], "l", lazy.layout.right(), desc="Move focus to right"),
    Key([mod], "j", lazy.layout.down(), desc="Move focus down"),
    Key([mod], "k", lazy.layout.up(), desc="Move focus up"),

    Key([mod, "shift"], "h", lazy.layout.shuffle_left(), desc="Move window to the left"),
    Key([mod, "shift"], "l", lazy.layout.shuffle_right(), desc="Move window to the right"),
    Key([mod, "shift"], "j", lazy.layout.shuffle_down(), desc="Move window down"),
    Key([mod, "shift"], "k", lazy.layout.shuffle_up(), desc="Move window up"),

    Key([mod, "control"], "h", lazy.layout.grow_left(), desc="Grow window to the left"),
    Key([mod, "control"], "l", lazy.layout.grow_right(), desc="Grow window to the right"),
    Key([mod, "control"], "j", lazy.layout.grow_down(), desc="Grow window down"),
    Key([mod, "control"], "k", lazy.layout.grow_up(), desc="Grow window up"),

    Key([mod], "e", lazy.spawn('emacsclient -c'), desc="Start emacs"),
    Key([mod], "f", lazy.window.toggle_fullscreen(), desc="Toggle fullscreen on the focused window"),
    Key([mod], "i", lazy.layout.grow(), desc="Grow the layout"),
    Key([mod], "m", lazy.layout.shrink(), desc="Shrink the layout"),
    Key([mod], "q", lazy.function(run_script, 'lock.sh'), desc="Set layout to normal size"),
    Key([mod, "shift"], "q", lazy.window.kill(), desc="Kill focused window"),
    Key([mod, "control"], "q", lazy.shutdown(), desc="Shutdown Qtile"),
    Key([mod], "r", lazy.spawn(rofi), desc="Spawn a command using a prompt widget"),
    Key([mod, "control"], "r", lazy.reload_config(), desc="Reload the config"),
    Key([mod], "t", lazy.window.toggle_floating(), desc="Toggle floating on the focused window"),

    Key([mod], "Tab", lazy.next_layout(), desc="Toggle between layouts"),
    Key([mod], "Return", lazy.spawn(terminal), desc="Launch terminal"),
    Key([mod, "shift"], "Return", lazy.layout.toggle_split(), desc="Toggle between split and unsplit sides of stack"),

    KeyChord([mod], "s", [
	Key([], "t", lazy.spawn(terminal)),
	Key([], "b", lazy.spawn(browser)),
	],
	name="Spawn"
    )
]

for vt in range(1, 8):
    keys.extend([
	Key(
	    ["control", "mod1"],
	    f"f{vt}",
	    lazy.core.change_vt(vt).when(func=lambda: qtile.core.name == "wayland"),
	    desc=f"Switch to VT{vt}",
	)
    ])

#+END_SRC

* Groups

#+BEGIN_SRC python
group_names = [i for i in "1234567890"]

#group_labels = ["DEV", "WWW", "SYS", "DOC", "COM", "6", "7", "8", "9", "MISC"]
group_labels = ["1: DEV", "2: WWW", "3: SYS", "4: DOC", "5: COM", "6", "7", "8", "9", "0: MISC"]
#group_labels = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"]

groups = []
for i in range(len(group_names)):
    groups.append(
	Group(
	    name=group_names[i],
	    label=group_labels[i],
	)
    )


for i in groups:
    keys.extend(
	[
	    Key(
		[mod],
		i.name,
		lazy.group[i.name].toscreen(),
		desc="Switch to group {}".format(i.name),
	    ),
	    Key(
		[mod, "shift"],
		i.name,
		lazy.window.togroup(i.name),
		desc="Switch to & move focused window to group {}".format(i.name),
	    ),
	]
    )


#+END_SRC

* Scratchpads

#+BEGIN_SRC python
spSettings = {
	"width":0.6,
	"height":0.8,
	"x":0.2,
	"y":0.1,
	"opacity":0.9,
}

groups.append(ScratchPad("sp", [
    DropDown("volume", "alacritty -e alsamixer", **spSettings),
    DropDown("term", "alacritty --class=scratchpad", **spSettings),
]))

keys.extend([
    Key([mod], "v", lazy.group['sp'].dropdown_toggle('volume')),
    Key([mod], "n", lazy.group['sp'].dropdown_toggle('term')),
])

#+END_SRC

* Layouts

#+BEGIN_SRC python

layoutSettings = {
    "margin":10,
    "border_focus": get_color(theme, "border_focus"),
    "border_normal": get_color(theme, "border_normal"),
}

layouts = [
    layout.MonadTall(ratio=0.65, align=layout.MonadTall._right, **layoutSettings),
    layout.MonadThreeCol(**layoutSettings),
    layout.MonadWide(**layoutSettings),
]

#+END_SRC

* Top Bar

#+BEGIN_SRC python
widget_defaults = dict (
    fontsize=18,
    padding=6,
    foreground=get_color(theme, "bar_foreground3"),
)

extension_defaults = widget_defaults.copy()

screens = [ Screen(
	wallpaper="~/Wallpapers/4860933.jpg",
	wallpaper_mode="stretch",
	top=bar.Bar(
	    [
		widget.Spacer(length=8),
		widget.CurrentLayoutIcon(),
		widget.TextBox("|", foreground=get_color(theme, "bar_foreground2")),
		widget.GroupBox(
		    highlight_method='line',
		    active=get_color(theme, "bar_foreground1"),
		    this_current_screen_border=get_color(theme, "bar_foreground2"),
		    inactive=get_color(theme, "bar_foreground4"),
		),
		widget.TextBox("|", foreground=get_color(theme,"bar_foreground2")),
		widget.Prompt(),
		widget.WindowName(),
		widget.Chord(
		    chords_colors={
			"launch": ("#ff0000", "#ffffff"),
		    },
		    name_transform=lambda name: name.upper(),
		),
		widget.Systray(),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.GenPollCommand(cmd="/home/auc/.config/qtile/scripts/disk_free.sh", fmt="{}", update_interval=5),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.CPU(format="{freq_current} GHz {load_percent}%"),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.Memory(format="{MemPercent}%", measure_mem='G'),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.Volume(fmt="{}"),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.Clock(format="%a, %b %d - %H:%M"),
		widget.Spacer(length=8),
	    ],
	    32,
	    foreground=get_color(theme, "bar_foreground3"),
	    background=get_color(theme, "bar_background1"),
	),
    ),
]

mouse = [
    Drag([mod], "Button1", lazy.window.set_position_floating(), start=lazy.window.get_position()),
    Drag([mod], "Button3", lazy.window.set_size_floating(), start=lazy.window.get_size()),
    Click([mod], "Button2", lazy.window.bring_to_front()),
]

dgroups_key_binder = None
dgroups_app_rules = []  # type: list
follow_mouse_focus = True
bring_front_click = False
floats_kept_above = True
cursor_warp = False
floating_layout = layout.Floating(
    float_rules=[
	,*layout.Floating.default_float_rules,
	Match(wm_class="confirmreset"),  # gitk
	Match(wm_class="makebranch"),  # gitk
	Match(wm_class="maketag"),  # gitk
	Match(wm_class="ssh-askpass"),  # ssh-askpass
	Match(title="branchdialog"),  # gitk
	Match(title="pinentry"),  # GPG key password entry
    ]
)
auto_fullscreen = True
focus_on_window_activation = "smart"
reconfigure_screens = True

auto_minimize = True

wl_input_rules = None

wmname = "LG3D"
#+END_SRC


qtile/config.py
import os
import subprocess

from libqtile import bar, layout, qtile, widget, hook
from libqtile.config import Click, Drag, Group, Key, Match, Screen, KeyChord, DropDown, ScratchPad
from libqtile.lazy import lazy
from libqtile.utils import guess_terminal
from colors import *

mod = "mod4"

theme=RofiOne
terminal = "alacritty"
browser = "brave" 
rofi = "rofi -show combi"

def run_script(qtile, script):
    run_command = os.path.expanduser('~/.config/qtile/scripts/' + script)
    subprocess.call(run_command)

@hook.subscribe.startup_once
def autostart_once():
    #lazy.spawn("picom -bc")
    #lazy.spawn("emacs --daemon")
    autostart_script = os.path.expanduser('~/.config/qtile/scripts/autostart.sh')
    subprocess.call(autostart_script)

keys = [
    Key([mod], "h", lazy.layout.left(), desc="Move focus to left"),
    Key([mod], "l", lazy.layout.right(), desc="Move focus to right"),
    Key([mod], "j", lazy.layout.down(), desc="Move focus down"),
    Key([mod], "k", lazy.layout.up(), desc="Move focus up"),

    Key([mod, "shift"], "h", lazy.layout.shuffle_left(), desc="Move window to the left"),
    Key([mod, "shift"], "l", lazy.layout.shuffle_right(), desc="Move window to the right"),
    Key([mod, "shift"], "j", lazy.layout.shuffle_down(), desc="Move window down"),
    Key([mod, "shift"], "k", lazy.layout.shuffle_up(), desc="Move window up"),

    Key([mod, "control"], "h", lazy.layout.grow_left(), desc="Grow window to the left"),
    Key([mod, "control"], "l", lazy.layout.grow_right(), desc="Grow window to the right"),
    Key([mod, "control"], "j", lazy.layout.grow_down(), desc="Grow window down"),
    Key([mod, "control"], "k", lazy.layout.grow_up(), desc="Grow window up"),

    Key([mod], "e", lazy.spawn('emacsclient -c'), desc="Start emacs"),
    Key([mod], "f", lazy.window.toggle_fullscreen(), desc="Toggle fullscreen on the focused window"),
    Key([mod], "i", lazy.layout.grow(), desc="Grow the layout"),
    Key([mod], "m", lazy.layout.shrink(), desc="Shrink the layout"),
    Key([mod], "q", lazy.function(run_script, 'lock.sh'), desc="Set layout to normal size"),
    Key([mod, "shift"], "q", lazy.window.kill(), desc="Kill focused window"),
    Key([mod, "control"], "q", lazy.shutdown(), desc="Shutdown Qtile"),
    Key([mod], "r", lazy.spawn(rofi), desc="Spawn a command using a prompt widget"),
    Key([mod, "control"], "r", lazy.reload_config(), desc="Reload the config"),
    Key([mod], "t", lazy.window.toggle_floating(), desc="Toggle floating on the focused window"),

    Key([mod], "Tab", lazy.next_layout(), desc="Toggle between layouts"),
    Key([mod], "Return", lazy.spawn(terminal), desc="Launch terminal"),
    Key([mod, "shift"], "Return", lazy.layout.toggle_split(), desc="Toggle between split and unsplit sides of stack"),

    KeyChord([mod], "s", [
	Key([], "t", lazy.spawn(terminal)),
	Key([], "b", lazy.spawn(browser)),
	],
	name="Spawn"
    )
]

for vt in range(1, 8):
    keys.extend([
	Key(
	    ["control", "mod1"],
	    f"f{vt}",
	    lazy.core.change_vt(vt).when(func=lambda: qtile.core.name == "wayland"),
	    desc=f"Switch to VT{vt}",
	)
    ])

group_names = [i for i in "1234567890"]

group_labels = ["DEV", "WWW", "SYS", "DOC", "COM", "6", "7", "8", "9", "MISC"]
#group_labels = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"]

groups = []
for i in range(len(group_names)):
    groups.append(
	Group(
	    name=group_names[i],
	    label=group_labels[i],
	)
    )


for i in groups:
    keys.extend(
	[
	    Key(
		[mod],
		i.name,
		lazy.group[i.name].toscreen(),
		desc="Switch to group {}".format(i.name),
	    ),
	    Key(
		[mod, "shift"],
		i.name,
		lazy.window.togroup(i.name),
		desc="Switch to & move focused window to group {}".format(i.name),
	    ),
	]
    )

spSettings = {
	"width":0.6,
	"height":0.8,
	"x":0.2,
	"y":0.1,
	"opacity":0.9,
}

groups.append(ScratchPad("sp", [
    DropDown("volume", "alacritty -e alsamixer", **spSettings),
    DropDown("term", "alacritty --class=scratchpad", **spSettings),
]))

keys.extend([
    Key([mod], "v", lazy.group['sp'].dropdown_toggle('volume')),
    Key([mod], "n", lazy.group['sp'].dropdown_toggle('term')),
])

layoutSettings = {
    "margin":10,
    "border_focus": get_color(theme, "border_focus"),
    "border_normal": get_color(theme, "border_normal"),
}

layouts = [
    layout.MonadTall(ratio=0.65, align=layout.MonadTall._right, **layoutSettings),
    layout.MonadThreeCol(**layoutSettings),
    layout.MonadWide(**layoutSettings),
]

widget_defaults = dict (
    fontsize=18,
    padding=6,
    foreground=get_color(theme, "bar_foreground3"),
)

extension_defaults = widget_defaults.copy()

screens = [ Screen(
	wallpaper="~/Wallpapers/4860933.jpg",
	wallpaper_mode="stretch",
	top=bar.Bar(
	    [
		widget.Spacer(length=8),
		widget.CurrentLayoutIcon(),
		widget.TextBox("|", foreground=get_color(theme, "bar_foreground2")),
		widget.GroupBox(
		    highlight_method='line',
		    active=get_color(theme, "bar_foreground1"),
		    this_current_screen_border=get_color(theme, "bar_foreground2"),
		    inactive=get_color(theme, "bar_foreground4"),
		),
		widget.TextBox("|", foreground=get_color(theme,"bar_foreground2")),
		widget.Prompt(),
		widget.WindowName(),
		widget.Chord(
		    chords_colors={
			"launch": ("#ff0000", "#ffffff"),
		    },
		    name_transform=lambda name: name.upper(),
		),
		widget.Systray(),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.GenPollCommand(cmd="/home/auc/.config/qtile/scripts/disk_free.sh", fmt="{}", update_interval=5),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.CPU(format="{freq_current} GHz {load_percent}%"),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.Memory(format="{MemPercent}%", measure_mem='G'),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.Volume(fmt="{}"),
		widget.TextBox("", foreground=get_color(theme, "bar_foreground2")),
		widget.Clock(format="%a, %b %d - %H:%M"),
		widget.Spacer(length=8),
	    ],
	    32,
	    foreground=get_color(theme, "bar_foreground3"),
	    background=get_color(theme, "bar_background1"),
	),
    ),
]

mouse = [
    Drag([mod], "Button1", lazy.window.set_position_floating(), start=lazy.window.get_position()),
    Drag([mod], "Button3", lazy.window.set_size_floating(), start=lazy.window.get_size()),
    Click([mod], "Button2", lazy.window.bring_to_front()),
]

dgroups_key_binder = None
dgroups_app_rules = []  # type: list
follow_mouse_focus = True
bring_front_click = False
floats_kept_above = True
cursor_warp = False
floating_layout = layout.Floating(
    float_rules=[
	*layout.Floating.default_float_rules,
	Match(wm_class="confirmreset"),  # gitk
	Match(wm_class="makebranch"),  # gitk
	Match(wm_class="maketag"),  # gitk
	Match(wm_class="ssh-askpass"),  # ssh-askpass
	Match(title="branchdialog"),  # gitk
	Match(title="pinentry"),  # GPG key password entry
    ]
)
auto_fullscreen = True
focus_on_window_activation = "smart"
reconfigure_screens = True

auto_minimize = True

wl_input_rules = None

wmname = "LG3D"
